types/src/timestamp.rs

```diff
@@ -1,48 +1,53 @@
 // TODO - remove once schemars stops causing warning.
 #![allow(clippy::field_reassign_with_default)]
 
+use core::{
+    ops::{Add, AddAssign, Div, Mul, Rem, Shl, Shr, Sub, SubAssign},
+    time::Duration,
+};
+#[cfg(any(feature = "std", test))]
 use std::{
     fmt::{self, Display, Formatter},
-    ops::{Add, AddAssign, Div, Mul, Rem},
     str::FromStr,
-    time::{Duration, SystemTime},
+    time::SystemTime,
 };
 
+#[cfg(feature = "datasize")]
 use datasize::DataSize;
-use derive_more::{Add, AddAssign, From, Shl, Shr, Sub, SubAssign};
+#[cfg(any(feature = "std", test))]
 use humantime::{DurationError, TimestampError};
-use once_cell::sync::Lazy;
-#[cfg(test)]
+#[cfg(any(feature = "testing", test))]
 use rand::Rng;
+#[cfg(feature = "json-schema")]
 use schemars::JsonSchema;
+#[cfg(any(feature = "std", test))]
 use serde::{de::Error as SerdeError, Deserialize, Deserializer, Serialize, Serializer};
 
-use casper_types::bytesrepr::{self, FromBytes, ToBytes};
-#[cfg(test)]
-use casper_types::testing::TestRng;
+#[cfg(any(feature = "std", test))]
+use crate::bytesrepr::{self, FromBytes, ToBytes};
 
-use crate::rpcs::docs::DocExample;
-
-static TIMESTAMP_EXAMPLE: Lazy<Timestamp> = Lazy::new(|| {
-    let example_str: &str = "2020-11-17T00:39:24.072Z";
-    Timestamp::from_str(example_str).unwrap()
-});
+#[cfg(any(feature = "testing", test))]
+use crate::testing::TestRng;
 
 /// A timestamp type, representing a concrete moment in time.
-#[derive(
-    DataSize, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Shr, Shl, JsonSchema,
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+#[cfg_attr(
+    feature = "json-schema",
+    derive(JsonSchema),
+    schemars(with = "String", description = "Timestamp formatted as per RFC 3339")
 )]
-#[serde(deny_unknown_fields)]
```

Comment:
Note: I'm not 100% sure about removing this, but since we're not deriving `Serialize` or `Deserialize`, I think it shouldn't have any effect? Or am I wrong?

Reply:
Should be fine to remove.